astropy/io/ascii/ui.py

```diff
@@ -220,7 +220,7 @@ def _guess(table, read_kwargs, format, fast_reader):
             reader.guessing = True
             return reader.read(table)
 
-        except (core.InconsistentTableError, ValueError, TypeError,
+        except (core.InconsistentTableError, ValueError, TypeError, AttributeError,
```

Comment:
ISTM a less obscure fix would be for the daophot reader to catch this exception and raise some more specific parsing error exception.